Docker.DotNet/Endpoints/MiscellaneousOperations.cs

```diff
@@ -64,32 +63,50 @@ public Task<Stream> MonitorEventsAsync(ContainerEventsParameters parameters, Can
             }
 
             JsonRequestContent<Config> data = parameters.Config == null ? null : new JsonRequestContent<Config>(parameters.Config, this.Client.JsonSerializer);
-                        
+
             IQueryString queryParameters = new QueryString<CommitContainerChangesParameters>(parameters);
             DockerApiResponse response = await this.Client.MakeRequestAsync(this.Client.NoErrorHandlers, HttpMethod.Post, "commit", queryParameters, data).ConfigureAwait(false);
             return this.Client.JsonSerializer.DeserializeObject<CommitContainerChangesResponse>(response.Body);
         }
 
         public Task<Stream> GetImageAsTarballAsync(string name, CancellationToken cancellationToken)
         {
-            if (string.IsNullOrEmpty(name))
+            return GetMultipleImagesAsTarballAsync(new[] { name }, cancellationToken);
+        }
+
+        public Task<Stream> GetMultipleImagesAsTarballAsync(string[] names, CancellationToken cancellationToken)
+        {
+            string path = "images/get";
+            EnumerableQueryString queryString = null;
+
+            if (names.Length > 0)
             {
-                throw new ArgumentNullException(nameof(name));
+                queryString = new EnumerableQueryString("names", names);
             }
 
-            string path = string.Format(CultureInfo.InvariantCulture, "images/{0}/get", name);
-            return this.Client.MakeRequestForStreamAsync(new[] {ImageOperations.NoSuchImageHandler}, HttpMethod.Get, path, null, null, cancellationToken);
+            return this.Client.MakeRequestForStreamAsync(new[] { ImageOperations.NoSuchImageHandler }, HttpMethod.Get, path, queryString, null, cancellationToken);
         }
 
         public Task LoadImageFromTarballAsync(Stream stream, CancellationToken cancellationToken)
         {
+            return LoadImageFromTarball(stream, new ImageLoadParameters { Quiet = true }, cancellationToken);
+        }
+
+        public Task<Stream> LoadImageFromTarballWithProgressAsync(Stream stream, CancellationToken cancellationToken)
+        {
+            return LoadImageFromTarball(stream, new ImageLoadParameters { Quiet = false }, cancellationToken);
+        }
+
+        public Task<Stream> LoadImageFromTarball(Stream stream, ImageLoadParameters parameters, CancellationToken cancellationToken)
```

Comment:
I see the answer to my above question about "progress" now. Lets chat about this.